src/components/ChatPopUp.tsx

```diff
@@ -162,41 +164,59 @@ export function ChatPopUp(props: ChatPopUpProps) {
   // to avoid message requests immediately on load while the popup is still "hidden"
   const [renderChat, setRenderChat] = useState(false);
 
+  // Set the initial value of the local storage flag for opening on load only if it doesn't already exist
+  const popupLocalStorageKey = "yextChatPopupOpenOnLoad";
+  if (window.localStorage.getItem(popupLocalStorageKey) === null) {
+    window.localStorage.setItem(
+      popupLocalStorageKey,
+      openOnLoad ? "true" : "false"
+    );
+  }
+  const openOnLoadLocalStorage =
+    window.localStorage.getItem(popupLocalStorageKey);
+
   // only fetch initial message when ChatPanel is closed on load (otherwise, it will be fetched in ChatPanel)
   useFetchInitialMessage(
     showInitialMessagePopUp ? console.error : handleError,
     false,
     (showUnreadNotification || showInitialMessagePopUp) &&
       !renderChat &&
-      !openOnLoad
+      (!openOnLoad || openOnLoadLocalStorage === "false")
   );
 
   useEffect(() => {
-    // Open panel on load if openOnLoad prop is true or there are messages in state (from browser storage)
-    if (!renderChat && (openOnLoad || messages.length > 1)) {
+    /* Open panel on load if: 
+      - openOnLoad prop is true or there are messages in state (from browser storage), and local storage flag not set 
+      - local storage flag openOnLoadLocalStorage is set and is true */
+    if (
+      !renderChat &&
+      ((messages.length > 1 && openOnLoadLocalStorage === "true") || openOnLoad)
```

Comment:
Can you extract this check and the conditions in L183-184 (which should actually check for messages.length as well) into one general boolean for opening the panel and use that in the two hooks?